extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleReflections.java

```diff
@@ -21,6 +21,9 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         // but it delegates all use to "oracle.jdbc.driver.OracleDriver" - which is also what's recommended by the docs.
         final String driverName = "oracle.jdbc.driver.OracleDriver";
         reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, driverName));
+
+        // for ldap style jdbc urls. e.g. jdbc:oracle:thin:@ldap://oid:5000/mydb1,cn=OracleContext,dc=myco,dc=com
+        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, "com.sun.jndi.ldap.LdapCtxFactory"));
```

Comment:
Ah I was thinking about it last night but didn't have time to play with it today. I'm not a big fan of always including the LDAP code but I have no idea how common this is.

I was wondering if maybe we should shoe in a new `DatabaseKind.ORACLE_LDAP` which would trigger this and behave exactly like an Oracle one.

Then in there, we would have a specific build step consuming JdbcDataSourceBuildItem and that would trigger the reflective registration if we have a datasource with this kind.

Reply:
you are right this might not be so frequent. it is used extensively in my company btu we have tons of oracle instances. so ldap there is used for addressability (hiding the real connection details).
did not seem a big deal to add this class to reflection though, even if the client app is not using the ldap style. but I cannot measure the impact.
a new database kind seems to give that url format too much importance. another option would be to add an optional build time property `support_ldap_style_urls`.

Reply:
For as much as I'd want to fully avoid JNDI lookups across the board, even without this patch it would still work in JVM mode.

I'd rather maintain JVM ~ native as similar as possible and I agree it's no big deal to register a single class.

Reply:
Yeah so I ended up with this proposal because I don't think we have database-specific properties right now, unfortunately.

Measuring the impact is easy: build a native executable with and without this patch and compare the size of the executable.

Reply:
> another option would be to add an optional build time property support_ldap_style_urls

That's an interesting idea, but would imply adding some additional configuration which applies to Oracle only.

I think I'd rather merge this as-is, accepting that LDAP support already existed (although not functional in native in this case), and then possibly/optionally follow up with a general solution for all of Quarkus code to disable JNDI.

Which frankly I thought we already did, as GraalVM in early days wasn't able to compile core relying on JNDI - it seems clealry this is no longer failing.

Reply:
Ok I'll measure.

With the new reflection directive:
 - 77476 (76M)	./target/quarkus-integration-test-jpa-oracle-999-SNAPSHOT-runner

Without the reflection directive (LDAP style broken):
 - 76820 (76M)	./target/quarkus-integration-test-jpa-oracle-999-SNAPSHOT-runner

Reply:
Please let's have some data before merging this. It's not hard to collect and we will be able to take an educated decision. I will be happy to merge it if the cost is not prohibitive.

Reply:
OK, perfect, let's merge then :). I was a bit concerned this type of things could bring half of Java with it including JAXB, AWT and Swing ;).